qiskit/algorithms/amplitude_amplifiers/amplification_problem.py

```diff
@@ -28,12 +28,13 @@ class AmplificationProblem:
     """
 
     def __init__(self,
-                 oracle: QuantumCircuit,
+                 oracle: Union[QuantumCircuit, Statevector],
```

Comment:
Could you update the type hints in the getters and setters of the `oracle` as well? And explain the behavior in the docstring if users pass a `Statevector`? I think how it's formulated right now is not well-suited for statevectors, but only for circuits 🙂

Reply:
That is a good point.   I checked the code again from that point　The following line accessing _oracle.num_qubits_ (many such lines) made me afraid that "it is not well-suited for statevectors".
https://github.com/Qiskit/qiskit-terra/blob/0fc83f44a6e80969875c738b2cee7bc33223e45f/qiskit/algorithms/amplitude_amplifiers/amplification_problem.py#L85

Then I recalled the original motivation for this pull request  -  the following line creating an AmplificationProblem instance with the oracle being a statevector. 
https://github.com/Qiskit/qiskit-terra/blob/0fc83f44a6e80969875c738b2cee7bc33223e45f/test/python/algorithms/test_grover.py#L100

Since this test works, _oracle.num_qubits_ should work for a statevector as well.   Actually I confirmed the statevector also has the _num_qubits_ property. 

So the way it is formulated right now is meant for both circuits and statevectors.

Reply:
I updated the type hints for the getter and setter.    I did not update the docstring since the behavior is the same for a statevector as a circuit.